### PR TITLE
feat(catalog/producer): add producers reference table + producer_id FK on 5 catalogues (#900)

### LINKED ISSUE #900: feat(catalog): add producers reference table + producer_id FK on hops/fermentables/yeasts/misc/equipment
## Context

Phase 1-3 of the catalogue refactor (#708 / #869, PRs #888 → #899) shipped 8 reference catalogues. The catalogue entities all carry TODO comments deferring the producer dimension to a dedicated **normalize-producers** PR — this issue tracks that PR.

## Important domain clarification: Producer ≠ Distributor

| | Producer (this PR) | Distributor (separate follow-up PR — see #901) |
|---|---|---|
| **Definition** | Who MAKES the product | Who SELLS the product to the brewer |
| **Cardinality** | 1 product = **1 producer** (unique brand owner) | 1 product = **N distributors** |
| **Examples for Citra** | Yakima Chief Hops only (HBC 394 trademark) | Brewferm, Hopt, MaltMiller, Northern Brewer… |
| **Examples for US-05** | Fermentis only | Brouwland, Brewferm, Saveur Bière, BrewDog Outpost… |
| **Data model** | Simple FK 1:1 on catalogue tables | M:N junction tables (one per catalogue) |
| **Powers** | Brand identity surface (logo, country, website) | Boutique outbound URLs, multi-vendor pricing |

This PR introduces **producers only**. The follow-up issue #901 (to be opened) introduces distributors and is the actual foundation for the boutique 'Acheter' button (issue #625).

## Why now

- Foundation for proper brand identity surfacing (manufacturer logos, country, official website on the catalogue picker)
- Closes the documented design debt in the 8 catalogue entities (search for "deferred to normalize-producers PR" in the codebase)
- Replaces the inline `yeasts.laboratory` (varchar) + `yeasts.product_id` (varchar) by a proper FK + a renamed `product_code` SKU column
- Non-blocking for the next big chantier (Recipe → catalogue FK refactor, separate PR)

## Scope

**1. New `producers` table** (~16 seeded entries spanning the 5 producer types):
- Fields: id, name (UNIQUE), type (enum CHECK: laboratory / maltster / hop_supplier / equipment_manufacturer / other), country (ISO 3166-1 alpha-2), website, notes (FR), timestamps
- Unique index on name + simple index on type

**2. Add nullable `producer_id` FK on 5 catalogue tables**:
- `hops.producer_id`
- `fermentables.producer_id`
- `yeasts.producer_id` (replaces `laboratory` varchar — data migration: lookup `laboratory` → `producer.name`)
- `yeasts.product_code` (renamed from `product_id` for clarity — it's a SKU, not a FK)
- `misc_templates.producer_id`
- `equipment_templates.producer_id`

All FKs use `ON DELETE SET NULL` so deleting a producer doesn't cascade-destroy ingredient rows.

**3. Standard catalogue files** (~14 total, mirroring the pattern set by PRs #888 → #899):
- entity + DTO + types + service + controller + module + migration + seed + 3 specs + runner
- Wiring updates to `catalog.module.ts` and `typeorm.config.ts`
- + ALTER TABLE migration on the 5 existing catalogues

**4. Seed (~16 producers)**: Wyeast, White Labs, Fermentis, Lallemand, Imperial Yeast, Briess, Weyermann, Best Malz, Castle, Crisp, Yakima Chief, BarthHaas, Hopsteiner, Grainfather/iMake, Klarstein, Anvil Brewing.

UUID range: `00000000-0000-4000-9000-8000-…` reserved for producers (`8` after the 7 catalogues `0`-`7`).

## Out of scope (explicitly)

- ❌ **Distributors** — separate follow-up issue #901 (M:N junctions, boutique URLs)
- ❌ M:N producers ↔ catalogues (over-engineering — there is genuinely only one producer per branded product)
- ❌ Recipe-X → catalogue FK refactor (separate PR — closes the recipe inline data debt)
- ❌ Yeast strain identity ("US-05 = WLP001 = 1056") — defer to dedicated `yeast_strains` PR if the need emerges
- ❌ Boutique outbound URL feature itself (issue #625 — separate frontend work depending on #901 first)

## Acceptance

- 5 catalogue tables have a `producer_id` column (nullable, FK to `producers.id`, ON DELETE SET NULL)
- `producers` table seeded with at least 15 entries spanning all 5 producer types
- Yeast catalogue's `laboratory` column dropped and migrated to `producer_id`
- `product_id` renamed to `product_code` on yeast
- New `GET /catalog/producers` + `/catalog/producers/:uuid` endpoints (read-only, JWT-protected, mirrors the 8 prior catalogue controllers)
- Tests: ~25 new (service / controller / seed specs)
- Lint clean, build green, all 584 existing tests still pass

### PR DESCRIPTION
**Pull request by benoit-bremaud**
## Summary

Closes **#900**. Foundation for the unified producer dimension that the 8 Phase 1-3 catalogue entities (#888 → #899) all deferred via TODO comments ("normalize-producers PR after Phase 3"). Adds a `producers` reference table seeded with **16 brand owners** spanning 5 producer types, plus a nullable `producer_id` FK on 5 catalogue tables.

## Important domain clarification: Producer ≠ Distributor

| | Producer (this PR) | Distributor (follow-up #901) |
|---|---|---|
| **Definition** | Who MAKES the product | Who SELLS it to the brewer |
| **Cardinality** | 1 product = **1 producer** | 1 product = **N distributors** |
| **Examples for Citra** | Yakima Chief Hops only | Brewferm, Hopt, MaltMiller, Northern Brewer… |
| **Data model** | FK 1:1 on catalogue tables | M:N junction tables (one per catalogue) |
| **Powers** | Brand identity surface | Boutique outbound URLs |

## Mode-prudent first cut (chosen on the fly during planning)

Yeast catalogue already has `laboratory` varchar + `product_id` varchar from PR #890. The "clean" scope would have dropped both and renamed `product_id` → `product_code` with a data migration. **This PR does NOT do that**: yeast carries 3 producer-related fields temporarily (producer_id FK + laboratory string + product_id string).

Trade-off chosen for:
- Minimal diff (5 catalogue files touched with 1 nullable column each, vs 7 if we'd done the full yeast cleanup)
- Zero risk on existing seeded yeast data (no DROP COLUMN, no data migration logic)
- Picker UI can be validated against the new FK before we drop the redundant columns
- Cleanup is a tiny follow-up PR (5 lines: drop laboratory, rename product_id → product_code)

## Schema (table `producers` — 8 columns total: 5 data + 3 system)

```sql
CREATE TABLE producers (
  id          varchar(36) PRIMARY KEY,
  name        varchar(120) NOT NULL UNIQUE,
  type        varchar(32) NOT NULL,    -- enum CHECK
  country     varchar(2),               -- ISO 3166-1 alpha-2 with CHECK
  website     varchar(255),
  notes       text,                     -- French, UI-facing
  created_at  datetime DEFAULT CURRENT_TIMESTAMP,
  updated_at  datetime DEFAULT CURRENT_TIMESTAMP,
  CONSTRAINT CHK_producers_type CHECK (type IN
    ('laboratory', 'maltster', 'hop_supplier',
     'equipment_manufacturer', 'other')),
  CONSTRAINT CHK_producers_country CHECK (country IS NULL
    OR (country GLOB '[A-Z][A-Z]' AND length(country) = 2))
);
```

Indices: UNIQUE on `name`, simple on `type`. The 5 catalogue tables get `IDX_<table>_producer_id` simple indices for picker-UI filter performance.

## FK additions (mode-prudent, no destructive change)

| Table | Column added | Behaviour |
|-------|--------------|-----------|
| `hops` | `producer_id varchar(36) NULL REFERENCES producers(id)` | ON DELETE SET NULL |
| `fermentables` | `producer_id` | ON DELETE SET NULL |
| `yeasts` | `producer_id` (alongside legacy laboratory + product_id) | ON DELETE SET NULL |
| `misc_templates` | `producer_id` | ON DELETE SET NULL |
| `equipment_templates` | `producer_id` | ON DELETE SET NULL |

## Endpoints

```
GET /catalog/producers          → ordered list
GET /catalog/producers/:uuid    → single entry, 404 on miss
```

## Seed (16 entries, idempotent loader)

5 **laboratories** (Wyeast Labs, White Labs, Fermentis, Lallemand, Imperial Yeast) — names match existing `yeasts.laboratory` strings VERBATIM so the future cleanup PR can lookup by equality.

5 **maltsters** (Briess, Weyermann, Best Malz, Castle Malting, Crisp Malting).

3 **hop suppliers** (Yakima Chief, BarthHaas, Hopsteiner).

3 **equipment manufacturers** (Grainfather/iMake, Klarstein, Anvil Brewing).

UUID range `00000000-0000-4000-9000-8000-…` reserved for producers (`8` after the 7 catalogues `0`-`7`).

## Tests (21 new, all green — 605 total in api package)

- **Service spec** — in-memory SQLite (4 tests): list ordered, getById happy/sad/edge
- **Controller spec** — `jest.spyOn` (4 tests): list happy/sad, getById happy/sad/edge (DTO wrapping)
- **Seed spec** — `assertCommonCatalogueSeederBehaviours` (4) + 6 catalogue-specific invariants (16 entries, deterministic UUID range, all 5 types covered, ISO alpha-2 country CHECK, French notes invariant, **yeast.laboratory string match assertion** for future cleanup)

## Out of scope (explicitly)

- ❌ Drop yeast.laboratory + rename product_id → product_code (follow-up cleanup PR after picker UX validation)
- ❌ Distributors / boutique foundation (separate issue #901, will be refined when boutique work begins)
- ❌ Backfill producer_id values on existing seeds of hops/fermentables/yeasts/misc/equipment (data PR — column is nullable, no breakage)
- ❌ Recipe-X → catalogue FK refactor (separate PR — closes the recipe inline data debt)
- ❌ M:N producers ↔ catalogues (genuinely 1 producer per branded product, no need)

## All review lessons from PR #888 → #899 baked in

- ✅ Boot-safe registration (ProducerOrmEntity in ormEntities)
- ✅ Migration consistency (CHECK constraints + indices match the hops/misc patterns)
- ✅ Module class suffix `*CatalogModule`
- ✅ NotFoundException message format matches every other catalogue
- ✅ `assertCommonCatalogueSeederBehaviours` shared helper (no duplication)
- ✅ French notes regex assertion
- ✅ Mode-prudent yeast strategy explicitly explained in the migration docstring + entity docstrings

## Checklist

- [x] Happy path + sad path + edge cases covered
- [x] `tsc --noEmit` passes (build green)
- [x] Build passes (`npm run build` clean)
- [x] Existing tests still green (605/607 — 2 skipped pre-existing, +21 producer)
- [x] No `any`, no inline styles introduced
- [x] Lint clean (only 1 pre-existing warning in `scan.service.ts`)

Closes #900

### CODE REVIEW

**Comment by benoit-bremaud** [comment]
@vitalikevin — Pour information : voici la **PR #902**, premier follow-up de la série catalogue (PR #888 → #899). 

**Ce qu'elle apporte concrètement**
On vient d'ajouter une nouvelle table de référence `producers` qui regroupe les **fabricants** des produits brassicoles : les laboratoires de levures (Wyeast, White Labs, Fermentis…), les malteries (Briess, Weyermann…), les négociants en houblon (Yakima Chief, BarthHaas…) et les constructeurs d'équipement (Grainfather, Klarstein, Anvil). 16 marques seedées au total.

**À noter — distinction importante** : un *producer* fabrique le produit (= 1 marque par produit). Un *distributor* le revend (= N boutiques peuvent vendre le même produit). Cette PR ne concerne QUE les producers. La table `distributors` (M:N) qui alimentera réellement le bouton "Acheter" de la future boutique est suivie séparément en issue #901, et sa spec sera affinée au moment où on attaquera la boutique.

**Ce que ça change pour le projet**
- 5 catalogues existants gagnent un FK optionnel `producer_id` (hops, fermentables, yeasts, misc, equipment)
- Le mobile pourra afficher "Citra — produit par Yakima Chief Hops 🇺🇸" au lieu d'un nom seul
- Fondation pour faire transparaître l'identité de marque dans le picker UI
- Mode prudent choisi : on ne touche pas aux colonnes existantes du yeast (`laboratory`, `product_id`) — un nettoyage tiny PR suivra une fois le picker validé

CI en cours, je préviens dès que tout est vert.

File: packages/api/src/database/migrations/1791000000000-AddProducersAndCatalogueFks.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      'misc_templates',
+      'yeasts',
+      'fermentables',
+      'hops',
+    ]) {
+      await queryRunner.query(`DROP INDEX "IDX_${table}_producer_id"`);
+    }
+    await queryRunner.query(`DROP INDEX "IDX_producers_type"`);
+    await queryRunner.query(`DROP INDEX "IDX_producers_name"`);
+    await queryRunner.query(`DROP TABLE "producers"`);
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/database/migrations/1791000000000-AddProducersAndCatalogueFks.ts:107]: Excellent catch P1, merci. Tu as raison, j'avais commenté "on ne rollback jamais en prod" dans le code mais c'est laisser une grenade dégoupillée pour rien. Fixé dans `9df4541` :

- Drop column index `IDX_<table>_producer_id` AVANT `ALTER TABLE <table> DROP COLUMN producer_id` (SQLite 3.35+ supporte DROP COLUMN, better-sqlite3 bundle SQLite 3.42+)
- Puis drop `IDX_producers_*` puis `DROP TABLE producers`

L'ordre est maintenant : pour chacun des 5 catalogues → drop index → drop column → puis drop indices producers → drop table producers. Plus aucune FK clause ne dangle après rollback, et le re-run du up() est propre. 605/607 tests toujours verts, aucune régression.

File: packages/api/src/catalog/producer/services/producer.service.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+
+import { InjectRepository } from '@nestjs/typeorm';
+import { ProducerOrmEntity } from '../entities/producer.orm.entity';
+import { Repository } from 'typeorm';
+
+@Injectable()
+export class ProducerService {
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/catalog/producer/services/producer.service.ts]: Bonne prise — fixed in 2d42349. Renamed file `producer.service.ts` → `producer-catalog.service.ts` and class `ProducerService` → `ProducerCatalogService`. Module + spec updated. Now consistent with HopCatalogService / YeastCatalogService / WaterCatalogService / EquipmentCatalogService / MiscCatalogService.

File: packages/api/src/catalog/producer/controllers/producer.controller.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ * Write endpoints (POST / PATCH / DELETE) are intentionally
+ * absent — the catalogue is operator-curated reference data,
+ * not user-managed content. Catalogue ingredients reference a
+ * producer by UUID through `producer_id` FK columns.
+ */
+@ApiTags('Catalog · Producers')
+@ApiBearerAuth('JWT-auth')
+@UseGuards(JwtAuthGuard)
+@Controller('catalog/producers')
+export class ProducerController {
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/catalog/producer/controllers/producer.controller.ts]: Same root cause as the service rename — fixed in 2d42349. Controller is now `ProducerCatalogController` in `producer-catalog.controller.ts`. Module + spec updated.

File: packages/api/src/catalog/hop/entities/hop.orm.entity.ts [hunk 4]
```diff
@@ -91,6 +91,18 @@ export class HopOrmEntity {
...
+   * Optional FK to the unified `producers` reference table
+   * (Issue #900). Identifies the brand owner — for trademarked
+   * varieties (Citra, Mosaic, Galaxy) this is the unique
+   * producer; for open varieties (Cascade, Saaz) this is the
+   * curated default. Nullable so legacy/future entries without
+   * an identified producer remain valid. ON DELETE SET NULL so
+   * deleting a producer never cascades onto the catalogue row.
+   */
+  @Column({ type: 'varchar', length: 36, nullable: true })
+  producer_id: string | null;
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/catalog/hop/entities/hop.orm.entity.ts:105]: Agreed — fixed in 2d42349. `HopDto` now exposes `producer_id` (uuid, nullable) with ApiProperty + fromEntity mapping. The picker can now read the FK from `GET /catalog/hops/:id`.

File: packages/api/src/catalog/fermentable/entities/fermentable.orm.entity.ts [hunk 5]
```diff
@@ -111,6 +111,17 @@ export class FermentableOrmEntity {
...
+  /**
+   * Optional FK to the unified `producers` reference table
+   * (Issue #900) — the maltster that produces this exact malt
+   * (Briess, Weyermann, Best Malz, Castle, Crisp…). Same malt
+   * type sold by multiple maltsters typically has slightly
+   * different chemistry, so each maltster's variant is its own
+   * catalogue row. Nullable + ON DELETE SET NULL.
+   */
+  @Column({ type: 'varchar', length: 36, nullable: true })
+  producer_id: string | null;
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/catalog/fermentable/entities/fermentable.orm.entity.ts:124]: Same root cause as the hop DTO comment — fixed in 2d42349. `FermentableDto` now exposes `producer_id` (uuid, nullable, description points to maltster brand).

File: packages/api/src/catalog/yeast/entities/yeast.orm.entity.ts [hunk 6]
```diff
@@ -113,6 +113,19 @@ export class YeastOrmEntity {
...
+   * (Issue #900) — the laboratory that brands this strain
+   * (Wyeast Labs, White Labs, Fermentis, Lallemand, Imperial
+   * Yeast). Mode-prudent first cut: this column is added
+   * ALONGSIDE the existing `laboratory` varchar (PR #890),
+   * not in replacement. The cleanup PR (drop `laboratory`,
+   * rename `product_id` → `product_code`) ships separately
+   * once picker UX is validated. Nullable + ON DELETE SET NULL.
+   */
+  @Column({ type: 'varchar', length: 36, nullable: true })
+  producer_id: string | null;
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/catalog/yeast/entities/yeast.orm.entity.ts:128]: Same root cause as the hop DTO comment — fixed in 2d42349. `YeastDto` now exposes `producer_id` (uuid, nullable). The description explicitly notes that mode-prudent keeps the legacy `laboratory` + `product_id` columns alongside until the cleanup PR.

File: packages/api/src/catalog/misc/entities/misc-template.orm.entity.ts [hunk 7]
```diff
@@ -74,6 +74,16 @@ export class MiscTemplateOrmEntity {
...
 
+  /**
+   * Optional FK to the unified `producers` reference table
+   * (Issue #900). Most miscs have a single producer (Servomyces
+   * = Lallemand, Whirlfloc = Kerry Group), commodity items
+   * (lactose, coriandre) leave it NULL. Nullable + ON DELETE
+   * SET NULL.
+   */
+  @Column({ type: 'varchar', length: 36, nullable: true })
+  producer_id: string | null;
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/catalog/misc/entities/misc-template.orm.entity.ts:87]: Same root cause as the hop DTO comment — fixed in 2d42349. `MiscTemplateDto` now exposes `producer_id` (uuid, nullable, description gives the Lallemand/Servomyces example).

File: packages/api/src/catalog/equipment/entities/equipment-template.orm.entity.ts [hunk 8]
```diff
@@ -96,6 +96,17 @@ export class EquipmentTemplateOrmEntity {
...
+  /**
+   * Optional FK to the unified `producers` reference table
+   * (Issue #900) — the equipment manufacturer (Grainfather/
+   * iMake, Klarstein, Anvil Brewing). Each equipment template
+   * has at most one manufacturer (1:1 relationship).
+   * Nullable + ON DELETE SET NULL — the kitchen-starter
+   * Casserole 5L for instance has no specific manufacturer.
+   */
+  @Column({ type: 'varchar', length: 36, nullable: true })
+  producer_id: string | null;
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/catalog/equipment/entities/equipment-template.orm.entity.ts:110]: Same root cause as the hop DTO comment — fixed in 2d42349. `EquipmentTemplateDto` now exposes `producer_id` (uuid, nullable, description gives Grainfather/Klarstein examples).

File: packages/api/src/catalog/hop/entities/hop.orm.entity.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/catalog/hop/entities/hop.orm.entity.ts:105]: Right — fixed in 2d42349. Added `@Index(['producer_id'])` on `HopOrmEntity`. Now matches the `@Index(['usage_type'])` and `@Index(['form'])` already on the class, and the migration's `IDX_hops_producer_id` is mirrored in the ORM metadata.

File: packages/api/src/catalog/fermentable/entities/fermentable.orm.entity.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/catalog/fermentable/entities/fermentable.orm.entity.ts:124]: Same root cause as the hop @Index comment — fixed in 2d42349. `@Index(['producer_id'])` added to `FermentableOrmEntity`.

File: packages/api/src/catalog/yeast/entities/yeast.orm.entity.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/catalog/yeast/entities/yeast.orm.entity.ts:128]: Same root cause as the hop @Index comment — fixed in 2d42349. `@Index(['producer_id'])` added to `YeastOrmEntity` (alongside the existing `@Index(['type'])`, `['form']`, `['laboratory']`).

File: packages/api/src/catalog/misc/entities/misc-template.orm.entity.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/catalog/misc/entities/misc-template.orm.entity.ts:87]: Same root cause as the hop @Index comment — fixed in 2d42349. `@Index(['producer_id'])` added to `MiscTemplateOrmEntity` (and Index added to the typeorm import block since it wasn't previously used on this entity).

File: packages/api/src/catalog/equipment/entities/equipment-template.orm.entity.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/catalog/equipment/entities/equipment-template.orm.entity.ts:110]: Same root cause as the hop @Index comment — fixed in 2d42349. `@Index(['producer_id'])` added to `EquipmentTemplateOrmEntity` (Index added to the typeorm import block since it wasn't previously used).

File: packages/api/src/database/seeds/producers-catalog.seed.ts [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+  id: string;
+  name: string;
+  type: ProducerType;
+  country: string | null;
+  website: string | null;
+  notes: string | null;
+}
+
+export const PRODUCERS_CATALOG_SEED: readonly ProducerSeed[] = [
+  // ─── Laboratories (yeast) ──────────────────────────────────────────
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/database/seeds/producers-catalog.seed.ts:38]: Good catch on the contract drift — fixed in 2d42349. Added a 17th seed entry: **Brouwland** (BE, type `other`). Brouwland is the historical Belgian generalist (founded 1976) — a distributor that also edits its own DIY kits and accessories. Doesn't fit any of the 4 specialised types (laboratory / maltster / hop_supplier / equipment_manufacturer), so `other` is the right home. Now every `ProducerType` enum value has at least one seed row.

File: packages/api/src/database/seeds/__tests__/producers-catalog.seed.spec.ts [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+    it('covers every ProducerType at least once', () => {
+      // The picker UI will offer a TYPE filter — every category
+      // must have at least one entry from day one so the filter
+      // chips never look empty.
+      const types = new Set(PRODUCERS_CATALOG_SEED.map((p) => p.type));
+      expect(types).toContain(ProducerType.Laboratory);
+      expect(types).toContain(ProducerType.Maltster);
+      expect(types).toContain(ProducerType.HopSupplier);
+      expect(types).toContain(ProducerType.EquipmentManufacturer);
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/database/seeds/__tests__/producers-catalog.seed.spec.ts:66]: Right, the assertion was incomplete. Fixed in 2d42349 — the spec now explicitly asserts `expect(types).toContain(ProducerType.Other)` alongside the 4 other types, plus the `exposes 17 curated catalogue entries` count assertion is bumped to 17. The regression this test guards is now actually tested.

File: packages/api/src/catalog/producer/entities/producer.orm.entity.ts [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+ * separate table introduced by issue #901 and power the
+ * boutique 'Acheter' button. Keep the two distinct.
+ *
+ * Referenced by 5 catalogue tables via nullable FK
+ * `producer_id` (ON DELETE SET NULL):
+ *   - hops
+ *   - fermentables
+ *   - yeasts (replaces the previous `laboratory` varchar +
+ *     `product_id` varchar columns; `product_code` is kept
+ *     as a separate non-FK SKU string)
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/catalog/producer/entities/producer.orm.entity.ts]: Tu as raison, le docblock était périmé par rapport au scope mode-prudent qu'on a finalement adopté. Fixed in 2d42349 — la section yeast du docblock dit maintenant explicitement "added ALONGSIDE the existing `laboratory` + `product_id` columns (PR #890), NOT in replacement" et précise que la cleanup PR ship séparément quand le picker UX sera validé.

File: packages/api/src/database/migrations/1791000000000-AddProducersAndCatalogueFks.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/database/migrations/1791000000000-AddProducersAndCatalogueFks.ts:107]: Doublon avec le commentaire P1 de Codex sur le même symptôme (FK clauses dangling après rollback). Approche choisie : DROP COLUMN sur les 5 catalogues AVANT le DROP TABLE producers (commit `9df4541`, posté avant que je voie ton commentaire). Du coup pas besoin de `PRAGMA foreign_keys = OFF` — une fois les colonnes supprimées, plus aucune FK ne pointe vers `producers` et le drop passe sans pragma. Effet net équivalent à ta suggestion : le rollback est maintenant réellement réversible.

**Comment by benoit-bremaud** [comment]
## Ready to merge

**CI** — all green on `2d42349`:
- api: pass (1m36s)
- SonarCloud Code Analysis: pass (42s)
- SonarCloud Scan: pass (2m44s)
- security-audit + changes: pass
- mobile-app / beer-encyclopedia / website: skipped (path-filtered, expected)

**Reviews** — 16 inline comments addressed (1 Codex + 15 Copilot):

**Codex P1** (`9df4541`):
- down() migration: drop `producer_id` columns BEFORE dropping `producers` table to avoid dangling FK clauses on rollback (SQLite 3.35+ DROP COLUMN, better-sqlite3 bundles 3.42+)

**Copilot 15 comments** (`2d42349`), grouped by theme:

| Group | # | Action |
|-------|---|--------|
| **A. DTO exposure** | 5 | Added `producer_id` field + ApiProperty + fromEntity mapping on HopDto, FermentableDto, YeastDto, MiscTemplateDto, EquipmentTemplateDto |
| **B. @Index decorator** | 5 | Added `@Index(['producer_id'])` on HopOrmEntity, FermentableOrmEntity, YeastOrmEntity, MiscTemplateOrmEntity, EquipmentTemplateOrmEntity (added Index import on misc + equipment) |
| **C. *Catalog* naming** | 2 | Renamed `ProducerService` → `ProducerCatalogService` and `ProducerController` → `ProducerCatalogController` (file + class + module + 2 specs) |
| **D. Other type coverage** | 2 | Added 17th seed entry **Brouwland** (BE, type=other) + spec assertion now covers ProducerType.Other + entry count bumped to 17 |
| **E. Producer entity docblock** | 1 | Fixed misleading "replaces previous laboratory + product_id" → now explicitly says mode-prudent keeps both alongside |
| **F. PRAGMA foreign_keys** | 1 | Duplicate of Codex P1 — same fix in 9df4541 (DROP COLUMN approach makes PRAGMA unnecessary) |

**Tests** — 605/607 green (2 pre-existing skipped), +21 from this PR, no regression.

**Commits**:
- `f74220a` — feat(catalog/producer): producers table + 5 catalogue FKs + 16 seed entries (~1093 LoC)
- `9df4541` — fix: Codex P1 (down migration FK dangle)
- `2d42349` — fix: Copilot review (15 comments — DTO + Index + naming + Other + docblock)

Per the merge gate convention, awaiting your explicit go to run `gh pr merge --squash --delete-branch`.
